### ISSUE TITLE
Installation Issues

### ISSUE THREAD

**Issue by richleyh** [asishrs/smartthings-ringalarmv2#31]
Hello!

   Thanks for updating this! I am having issues with installation. I have completely removed previous installation, and have meticulously followed the instruction for installation of this version [twice], but regardless, there is no connection between SmartThings and Ring Alarm. I initially tried with New Smartthing app, but reverted to classic. Doesn't seem to make a difference which app I use. Doesn't work in Webcore. Not sure what I've done wrong...? Is there a check you could suggest that would help me isolate the problem? For example, I seem to remember from version 2 a means of testing the API? I'm not very fluent with any of this, but I had version 2 and loved it. Thanks for any help you can offer!

**Comment by asishrs** [comment]
Can you share logs from your ST IDE for Device Handler and App? 

Also, confirm you are using latest version of Lambda https://github.com/asishrs/smartthings-ringalarmv2/releases/tag/v1.3.3 as well as the Device handler https://github.com/asishrs/smartthings

**Comment by richleyh** [comment]
Hi!

   Thanks so much for your prompt response and help with this. I really appreciate your help, and want to give you the full context of my setup. 
   I switched to new ST app due to inconsistent presence sensor performance, as well as the inevitable sunsetting of that app, some time back. We have 4 in our family, and the dream is to have our Ring alarm driven by the presence sensors through Webcore (for example, when my daughter arrives, alarm disarms, or when everyone is gone, arm system to Armed Away). Each of the 4 users has a discrete ST account.
   For this install, I have been using only the ST Classic app, because you have mentioned this has only been tested with ST Classic. When I enter the API invoke URL and API key, Ring account login info, etc. into the SmartApp, all is well, but as soon as I click "Done", and return to my list of installed smart apps, and then re-enter the Ring Alarm Manager SmartApp to confirm the continued existence of my input information, all of my inputs are gone. It doesn't seem to retain the information I've entered. Further, when I then open the SmartApp in the new ST app, I see multiple instances of the SmartApp installed, and they do contain my invoke URL, API key, and Ring account login info. I suspect, but don't know, that the number of instances of the SmartApp appearing in the new ST app is directly related to the number of attempts I've made to enter the information into the classic app.
    Given this, I concluded that perhaps the new ST app was interferring, so I completely un-installed the new ST app. However, the problem of not retaining my inputs persists, despite my deleting the new ST completely from my phone. 
   My live logging file is attached. I have removed and installed the DHs and SmartApp several times. I have also ensured use of the most current "deployment.zip" by deleting and re-creating the bucket a couple of times. I've also deleted and rebuilt the Stack a couple of times.  Everything appears to install correctly as I follow your steps, and I receive the confirmations you mention. 
   Again, I really appreciate your assistance with this. Thank you!

Rich
<img width="1174" alt="Screen Shot 2020-02-01 at 7 16 05 AM" src="https://user-images.githubusercontent.com/49592858/73592925-3532f700-44c5-11ea-95ec-0cef63978fc2.png">

**Comment by asishrs** [comment]
>> When I enter the API invoke URL and API key, Ring account login info, etc. into the SmartApp, all is well, but as soon as I click "Done", and return to my list of installed smart apps, and then re-enter the Ring Alarm Manager SmartApp to confirm the continued existence of my input information, all of my inputs are gone.

This is the default behavior of ST Apps (at least in iOS devices). If you are switching between apps ion your smartphone, the ST application won't keep the data. I recommend putting some default values on other fields and tap on done before you switch the application. Next time when you switch the app back to ST app, you can edit the value which was given earlier. 

>> Given this, I concluded that perhaps the new ST app was interferring, so I completely un-installed the new ST app. However, the problem of not retaining my inputs persists, despite my deleting the new ST completely from my phone.

As far as I know, the old and new versions will not impact each other.

Finally, based on your logs it looks like the AWS API information is not proper in your ST smartapp.

**Comment by asishrs** [comment]
Are you still looking for help with this?

**Comment by richleyh** [comment]
Thanks so much for following up. I have done the install several times without success. Finally, I tried to simply retreat to Version 2, but this too proved impossible, as some of the menu structures at AWS have changed, and the old install instructions no longer work, and I don't know enough about all of this to find my way around the new AWS menus and do what needs to be done. I will try to uninstall everything and re-install the new version again today, but I am at the point of simply accepting that I will need to manually control our alarm for now. I will let you know how it goes today. Thanks again!

**Comment by asishrs** [comment]
I have automated almost every steps required to deploy the AWS lambda api in latest version. I suggest using that for completing the api part.

**Comment by richleyh** [comment]
Yes, thank you. It seems straightforward enough. I am working on it as I write this. I will let you know how it goes. Thanks again so much for your support!

Rich

**Comment by richleyh** [comment]
When I get to this point of the install (I don’t have 2FA activated on my account): 
f you said NO to step 2.1, proceed as below.

Tap on Account Status to see the Ring Account Connection Status. If the connection is successful, you can see the ZID and Location.

...and tap on "Ring account status", (under the heading “Two Factor Authentication”, I see this:



   
The spinning circle then disappears, and nothing has happened. I don’t see ZID or Location. Based upon how it appears in the app, it looks like this would only be functional if I had 2FA turned on. Yet your instructions say, “If you said “no” to step 2.1 (that is, I do NOT have 2FA activated), tap on Account Status…etc”. Perhaps I am misreading something? I have confirmed multiple times that my Ring account username and password are correct.

Thanks for you continued support with this!

**Comment by richleyh** [comment]
Here are logs and screenshots. I have cut and pasted the invoke URL and API Key into the smart app to minimize the opportunity for a transcription error. I believe I have followed the instructions precisely. Thanks aging for looking at this with me!

**Comment by richleyh** [comment]
Just stumbled upon this while looking at the Lambda:

**Comment by dmoses1969** [comment]
asishrs, any previous versions of this integration must be removed prior to installing version 3 correct? Just want to make sure.

**Comment by asishrs** [comment]
If you want, you can keep old version for AWS lambda. Regardless of this, you need to use the updated Lambda url and api key with new ST app.

- Asish

**Comment by dmoses1969** [comment]
> If you want, you can keep old version for AWS lambda. Regardless of this, you need to use the updated Lambda url and api key with new ST app. - Asish
> […](#)
> On Feb 10, 2020, 8:51 AM -0700, dmoses1969 ***@***.***>, wrote: asishrs, any previous versions of this integration must be removed prior to installing version 3 correct? Just want to make sure. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

Thanks for the quick response. I went ahead and deleted everything I had and completed the steps for V3. 
I don't see all my Ring devices in SmartThings. Should I? In the previous versions we had to specify in the groovy file the number of each devices that we had. Do we still have to do that?

Thanks

**Comment by asishrs** [comment]
No need to add the count as that is part of the app now. Did you update the ST app to use the device handler as well.

- Asish

**Comment by dmoses1969** [comment]
Yes I did. Maybe I should remove everything and try again.

**Comment by dmoses1969** [comment]
> No need to add the count as that is part of the app now. Did you update the ST app to use the device handler as well. - Asish
> […](#)
> On Feb 10, 2020, 9:51 AM -0700, dmoses1969 ***@***.***>, wrote: > If you want, you can keep old version for AWS lambda. Regardless of this, you need to use the updated Lambda url and api key with new ST app. - Asish > … > On Feb 10, 2020, 8:51 AM -0700, dmoses1969 @.***>, wrote: asishrs, any previous versions of this integration must be removed prior to installing version 3 correct? Just want to make sure. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Thanks for the quick response. I went ahead and deleted everything I had and completed the steps for V3. I don't see all my Ring devices in SmartThings. Should I? In the previous versions we had to specify in the groovy file the number of each devices that we had. Do we still have to do that? Thanks — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

I found the issue but not sure how to fix it. I have 2 Ring Alarm Locations. My house and my mother's house. The Ringdevices I see listed in my SmartThings App are devices in my mother's house. Not sure what to do with it now.

**Comment by dmoses1969** [comment]
I have my Mothers Ring System as a shared location in my Ring setup. When I logged in with my credentials in the ST App, It looks like it used my "shared" location to sync with. Would it be possible to add a location setting to the app so only the selected Location would sync with SmartThings?

**Comment by asishrs** [comment]
That’s an interesting situation. Let me check the apis to see if that can be configurable.

**Comment by dmoses1969** [comment]
@asishrs Thanks for looking into it. 
  
I removed the 2nd location from my account temporarily to test this. I went thru the steps again this morning and I am connected and seems to be working. The only thing I do not see is my Smoke/CO Alarms. I will try adding my 2nd location back to my Ring account and see what happens. Hopefully it will not add those devices.

**Comment by asishrs** [comment]
If you add the second location now on your ring account, it should not impact you ST setup. If you remove the ST setup and add it after adding the second location, it may land in same situation. 

I completed the server side changes for multiple location support. I need to update the ST app for this. I can push a test build for this by end of this week, appreciate if you can help me to test it.

**Comment by dmoses1969** [comment]
Again, thanks for working on this. I can test whenever your ready. The only other issue I see is I still am not seeing my Smoke/CO detectors in ST.

**Comment by asishrs** [comment]
Do you have this one https://shop.ring.com/products/alarm-smoke-co-listener

**Comment by dmoses1969** [comment]
No, I have 6 of these. https://shop.ring.com/products/first-alert-smoke-co-alarm-zcombo-g-z-wave

**Comment by dmoses1969** [comment]
Not sure if this would help in any way but at one time I had them connected to ST and they worked fine. I moved them to Ring for home insurance discount.  Here some info on what they can report...
https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zwave-smoke-alarm.src/zwave-smoke-alarm.groovy

**Comment by asishrs** [comment]
> No, I have 6 of these. https://shop.ring.com/products/first-alert-smoke-co-alarm-zcombo-g-z-wave

Currently, the ST app doesn't support this. I can add the support in the upcoming build.
